src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -335,8 +336,32 @@ static ss::future<response_ptr> do_handle(alter_op_context& octx) {
         return octx.rctx.respond(std::move(octx.response));
     }
 
-    if (!octx.rctx.authorized(
-          security::acl_operation::alter, security::default_cluster_name)) {
+    auto authz = octx.rctx.authorized(
+      security::acl_operation::alter, security::default_cluster_name);
+
+    const auto additional_resources = [&octx]() {
+        std::vector<model::topic> topics;
+        topics.reserve(octx.request.data.topics.size());
+        std::transform(
+          octx.request.data.topics.begin(),
+          octx.request.data.topics.end(),
+          std::back_inserter(topics),
+          [](const reassignable_topic& t) { return t.name; });
+
+        return topics;
+    };
+
+    if (!octx.rctx.audit(
```

Comment:
Thinking out loud, if `audit` took a std::function instead of parameters to build an event then we could avoid calling the logic above in the event auditing is disabled, which is a common case

Reply:
Great idea.

Other idea is a variadic template that passes the arguments to the correct `make` in utils.

I would love to _not_ expose the schemas if possible to calling apps, but that may require more re-work.

Reply:
Yeah actually that is what i did for `audit_manager.h` , i'm in favor of doing something like this

Reply:
going to push shortly (without this). will get to it soon